alica_test_utility/include/alica/test/FailBehaviour.hpp

```diff
@@ -0,0 +1,36 @@
+#pragma once
+
+#include "MockUpBehaviour.h"
+
+#include <cstdint>
+
+namespace alica::test
+{
+template <uint32_t iterationsBeforeFail>
+class FailBehaviour : public MockUpBehaviour
```

Comment:
can merge this and success behaviour and do success or failure based on constructor argument,